src/test/java/seedu/rc4hdb/logic/commands/modelcommands/FilterCommandTest.java

```diff
@@ -0,0 +1,127 @@
+package seedu.rc4hdb.logic.commands.modelcommands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.rc4hdb.commons.core.Messages.MESSAGE_RESIDENTS_LISTED_OVERVIEW;
+import static seedu.rc4hdb.logic.commands.modelcommands.ModelCommandTestUtil.assertCommandSuccess;
+import static seedu.rc4hdb.testutil.TypicalResidents.ALICE;
+import static seedu.rc4hdb.testutil.TypicalResidents.AMY;
+import static seedu.rc4hdb.testutil.TypicalResidents.getTypicalResidentBook;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.rc4hdb.model.Model;
+import seedu.rc4hdb.model.ModelManager;
+import seedu.rc4hdb.model.UserPrefs;
+import seedu.rc4hdb.model.resident.ResidentDescriptor;
+import seedu.rc4hdb.model.resident.predicates.AttributesMatchKeywordsPredicate;
+import seedu.rc4hdb.testutil.ResidentDescriptorBuilder;
+
+public class FilterCommandTest {
+    private Model model = new ModelManager(getTypicalResidentBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalResidentBook(), new UserPrefs());
+
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredList_success() {
+        String expectedMessage = String.format(MESSAGE_RESIDENTS_LISTED_OVERVIEW, 1);
+        AttributesMatchKeywordsPredicate predicate =
+                new AttributesMatchKeywordsPredicate(new ResidentDescriptorBuilder(ALICE).build());
+        FilterCommand command = new FilterCommand(new ResidentDescriptorBuilder(ALICE).build());
+        expectedModel.updateFilteredResidentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE), model.getFilteredResidentList());
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        ResidentDescriptor descriptor = new ResidentDescriptorBuilder().withName(ALICE.getName().toString())
```

Comment:
Maybe we could overload the `withName` method to take in `Name` too.
Same goes for the other `with...` methods.

Let's just keep this in view for next time so that making tests will go slightly quicker.

Reply:
I actually had it in a separate filter descriptor builder method. But I deleted it since we already had a resident builder. Did not think of overloading it haha that's a pretty good idea. Maybe we can do that soon after 1.2